src/Core/src/Credentials/ContainerProvider.php

```diff
@@ -77,4 +100,78 @@ public function getCredentials(Configuration $configuration): ?Credentials
             Credentials::adjustExpireDate(new \DateTimeImmutable($result['Expiration']), $date)
         );
     }
+
+    /**
+     * Checks if the provided IP address is a loopback address.
+     *
+     * @param string $host the host address to check
+     *
+     * @return bool true if the IP is a loopback address, false otherwise
+     */
+    public static function isLoopBackAddress(string $host)
+    {
+        // Validate that the input is a valid IP address
+        if (!filter_var($host, \FILTER_VALIDATE_IP)) {
+            return false;
+        }
+
+        // Convert the IP address to binary format
+        $packedIp = inet_pton($host);
+
+        // Check if the IP is in the 127.0.0.0/8 range
+        if (4 === \strlen($packedIp)) {
+            return 127 === \ord($packedIp[0]);
+        }
+
+        // Check if the IP is ::1
+        if (16 === \strlen($packedIp)) {
+            return $packedIp === inet_pton('::1');
+        }
+
+        // Unknown IP format
+        return false;
+    }
+
+    private function getFullUri(Configuration $configuration): ?string
+    {
+        $relativeUri = $configuration->get(Configuration::OPTION_CONTAINER_CREDENTIALS_RELATIVE_URI);
+        $fullUri = $configuration->get(Configuration::OPTION_POD_IDENTITY_CREDENTIALS_FULL_URI);
```

Comment:
can be moved at line 144 (no need to instanciate the `fullUri` variable when the `relativeUri` is not null